### DISCUSSION TITLE
Oracle Advanced Queue and custom type in Micronaut

### DISCUSSION THREAD

**Discussion by MarkoDju** [micronaut-projects/micronaut-core#7472] [General] [answered]
### Issue description

Hello!

I want to use Oracle Advanced Queueing and use a custom message type. I have found an example in Spring ([https://blog.javaforge.net/post/30858904340/oracle-advanced-queuing-spring-custom-types](url) where the _DefaultMessageListenerContainer_ class is extended and the _createConsumer_ method overridden. The method uses the _ORADataFactory_ needed for the message type mapping from an Oracle type to a Java class.

Can this be done in Micronaut? The JMS Listener and consumer are both defined using annotations and I cannot seem to find a way to set  the ORADataFactory.

**Comment by sdelamo** [comment]
I've moved to a discussion.

**Comment by MarkoDju** [comment, marked as answer]
I solved my problem with some hacking of existing Micronaut classes ([https://github.com/MarkoDju/aq-demo](url)). I replaced JMSListenerContainerFactory, JMSListenerContainer and DefaultSerializerDeserializer classes with a hacked version to make it work.
There was some guessing involved, but my demo app works. 

I hope support for custom types in AQ will be added to Micronaut in future releases.
